server/datastore/mysql/cron_stats.go

```diff
@@ -2,26 +2,56 @@ package mysql
 
 import (
 	"context"
-	"database/sql"
 
 	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 	"github.com/jmoiron/sqlx"
 )
 
-func (ds *Datastore) GetLatestCronStats(ctx context.Context, name string) (fleet.CronStats, error) {
-	stmt := `SELECT id, name, instance, created_at, updated_at, stats_type, status FROM cron_stats WHERE name = ? ORDER BY created_at DESC LIMIT 1`
-
-	var res fleet.CronStats
-	err := sqlx.GetContext(ctx, ds.reader, &res, stmt, name)
-	switch {
-	case err == sql.ErrNoRows:
-		return fleet.CronStats{}, nil
-	case err != nil:
-		return fleet.CronStats{}, ctxerr.Wrap(ctx, err, "select cron stats")
-	default:
-		return res, nil
+// GetLatestCronStats returns a slice of no more than two cron stats records, where index 0 (if
+// present) is the most recently created scheduled run, and index 1 (if present) represents a
+// triggered run that is currently pending.
+func (ds *Datastore) GetLatestCronStats(ctx context.Context, name string) ([]fleet.CronStats, error) {
+	stmt := `
+SELECT
```

Comment:
Nitpick, you can omit the outer select and aliases

```
(
    SELECT
    ...
)
UNION
(
    SELECT
    ...
)
```

Reply:
Originally I tried to make that work, but that doesn't allow the inner `ORDER BY ... LIMIT 1` statements in `cs1` and `cs2`

Reply:
I tried it and it works. Note that it's still wrapped in a pair of parenthesis ...

Reply:
Ah, I wasn't using the parenthesis :)

Thanks!